code/src/cljs/sixsq/nuvla/ui/credential/events.cljs

```diff
@@ -0,0 +1,200 @@
+      (s/explain form-spec credential)
+      (assoc db ::spec/form-valid? valid?))))
+
+
+(reg-event-db
+  ::validate-minio-credential-form
+  (fn [db [_]]
+    (let [form-spec      ::spec/minio-credential
+          credential     (get db ::spec/credential)
+          validate-form? (get db ::spec/validate-form?)
+          valid?         (if validate-form? (if (nil? form-spec) true (s/valid? form-spec credential)) true)]
+      (s/explain form-spec credential)
+      (assoc db ::spec/form-valid? valid?))))
+
+
+(reg-event-db
+  ::set-validate-form?
+  (fn [db [_ validate-form?]]
+    (assoc db ::spec/validate-form? validate-form?)))
+
+
+; Set the spec to apply for form validation
+
+(reg-event-db
+  ::set-form-spec
+  (fn [db [_ form-spec]]
+    (assoc db ::spec/form-spec form-spec)))
+
+
+(reg-event-db
+  ::active-input
+  (fn [db [_ input-name]]
+    (assoc db ::spec/active-input input-name)))
+
+
+(reg-event-db
+  ::form-invalid
+  (fn [db [_]]
+    (assoc db ::spec/form-valid? false)))
+
+
+(reg-event-db
+  ::form-valid
+  (fn [db [_]]
+    (assoc db ::spec/form-valid? true)))
+
+
+(reg-event-db
+  ::set-credential
+  (fn [db [_ credential]]
+    (assoc db ::spec/credential credential)))
+
+
+(reg-event-db
+  ::set-credentials
+  (fn [db [_ credentials]]
+    (assoc db ::spec/credentials credentials)))
+
+
+(reg-event-fx
+  ::get-credentials
+  (fn [{{:keys [::client-spec/client] :as db} :db} [_]]
+    (when client
+      (let []
+        {:db                             (assoc db ::spec/completed? false)
+         ::credential-fx/get-credentials [client #(dispatch [::set-credentials %])]}))))
+
+
+(reg-event-fx
+  ::edit-credential
+  (fn [{{:keys [::spec/credential ::client-spec/client] :as db} :db :as cofx} [_]]
+    (log/infof "credential: %s" credential)
+    (let [id             (:id credential)
+          new-credential (utils/db->new-credential db)]
+      (log/infof "new-credential: %s" new-credential)
+      (if (nil? id)
+        {:db               db
+         ::cimi-api-fx/add [client "credential" new-credential
+                            #(if (instance? js/Error %)
+                               (let [{:keys [status message]} (response/parse-ex-info %)]
+                                 (dispatch [::messages-events/add
+                                            {:header  (cond-> (str "error editing " id)
+                                                              status (str " (" status ")"))
+                                             :content message
+                                             :type    :error}]))
+                               (do (dispatch [::cimi-detail-events/get (:id %)])
+                                   (dispatch [::close-credential-modal])
+                                   ;(dispatch [::set-module sanitized-module])
+                                   ;(dispatch [::main-events/changes-protection? false])
```

Comment:
Work in progress? After creating a credential, the warning of changes-protection? is visible.